### PR TITLE
Member popups

### PR DESCRIPTION
**Pull request by janbaykara**
At the closest zoom levels, allow interaction with individual members.

## Description

- Markers are clickable and display a popup with useful information
- Data sources now have designated fields (which are pre-emptively guessed on creation and can later be edited in the inspector) for phone, email, address, full name, first name, last name — imports will extract that data to be used for structured querying and data display

## Motivation and Context

CRMs are generally poor at contextualising member data in political context. They are primarily useful for broadcast comms, but what if you want to work in a different way?

Future iterations of this feature could involve segmenting members by tag or other fields — for example for wall-charting. You could then visualise this in political/geographic context and then plan outreach to specific members from the map.

## How Can It Be Tested?

Open a map report with a members list, and zoom in towards a ward. Eventually a marker will present itself and, when clicked, will display a pop up with useful, actionable information.

If you've already got a data source imported, then you'll need to inspect it and define the columns for things like phone / email, and then re-import, if you want to see it all in the popups.

## How Will This Be Deployed?

Standard.

## Screenshots (if appropriate):

Screenshot 2024-03-20 at 22 56 47

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx [hunk 1]
```diff
@@ -383,101 +386,103 @@ export default function Page({
...
+            >
+              <div className='text-hSm'>Connection details</div>
+              <FormField
+                control={form.control}
+                name="airtable.apiKey"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormLabel>Airtable access token</FormLabel>
+                    <FormControl>
+                      {/* @ts-ignore */}
```
Comment (joaquimds) [inline nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx:403]: suggestion: give an explanation for when linters are disabled

**Review by joaquimds — APPROVED** [review]
#lgtm

**Comment by joaquimds** [comment]
There's a CSS nesting error:

```
(97:7) Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.
See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting

Import trace for requested module:
./src/app/globals.css.webpack[javascript/auto]!=!./node_modules/next/dist/build/webpack/loaders/css-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[2]!./node_modules/next/dist/build/webpack/loaders/postcss-loader/src/index.js??ruleSet[1].rules[11].oneOf[12].use[3]!./src/app/globals.css
./src/app/globals.css
```

**Comment by janbaykara** [comment]
Closing to merge with #32

**Pull request closed without merging**